src/Lean/Server/FileWorker.lean

```diff
@@ -281,7 +283,6 @@ section Updates
     let initHeaderStx := (← get).initHeaderStx
     let (newHeaderStx, newMpState, _) ← Parser.parseHeader newMeta.mkInputContext
     let cancelTk ← CancelToken.new
-    -- Wait for at least one snapshot from the old doc, we don't want to unnecessarily re-run `print-paths`
```

Comment:
This comment was misleading, we only call `compileHeader` once in `initializeWorker`.

To be fair, I'm not sure how this even works at all.  If I quickly change `import Lean` to `import Lea` and back to `import Lean`, then `cmdSnaps` should always be `AsyncList.delayed <error>` from then on.  Even if we get back to the original header syntax, we'd just propagate the exception when unwrapping the head task.

But I couldn't turn this into an actual bug during (my limited) testing.

Reply:
> This comment was misleading, we only call `compileHeader` once in `initializeWorker`.

In a single worker process, yes, but thanks to worker restart we could quickly end up with parallel `print-paths` processes... which I apparently forgot about when moving the `exit` into a new task independent of `headSnapTask`. Which also means that in your example the worker is restarted each time, avoiding the bug. I only vaguely remember that I went through multiple revisions of where to put the `exit`.

Reply:
Thank you for the clarification!  That Lake continues to run even after the worker has stopped is a bigger issue, see also https://github.com/leanprover/lean4/issues/1438#issuecomment-1207354919